src/schedule/SchedulePing.ts

```diff
@@ -12,25 +12,50 @@ enum StartJobsStatus {
 export class SchedulePing {
   private handle?: NodeJS.Timeout;
   private startJobsStatus: StartJobsStatus = StartJobsStatus.notStarted;
+  private readonly maxPingRetries: number;
+  private readonly retryIntervalMs: number;
 
   constructor(
     private readonly schedulesRepository: SchedulesRepository,
     private readonly logger: Logger,
     private readonly interval: number,
     private readonly startAllJobs: () => Promise<void>,
-  ) {}
+    maxPingRetries: number = 1,
```

Comment:
Shouldn't these new fields also be `private readonly`? I guess then you wouldn't need to declare them in lines 15 and 16.